src/clients/ntp-client_proposal.rb

```diff
@@ -370,8 +385,13 @@ def Write(params)
 
       return :success if params["write_only"]
 
-      # Only if network is running try to synchronize the ntp server
+      # Only if network is running try to synchronize
+      # the ntp server.
       if NetworkService.isNetworkRunning && !Service.Active(NtpClient.service_name)
+        if !select_ntp_server && ntp_server.empty?
+          # Trying first server in the list
+          ntp_server = NtpClient.GetUsedNtpServers.first unless NtpClient.GetUsedNtpServers.nil?
```

Comment:
and same here with list of servers. If you have multiple servers and it sync only against one and that one failing, it fails whole, which is strange and not how works list of ntp servers. Also time will be less precise then with multiple sources.

Reply:
Yes, I have updated it. Now each ntp server will be tried until one has succeeded.